cms/template_tags_test.py

```diff
@@ -20,3 +22,23 @@ def test_image_version_url():
         result_url
         == f"/images/{expected_signature}/{image_id}/{image_filter}/?v={file_hash}"
     )
+
+
+@pytest.mark.parametrize(
+    "start, end",
+    [
+        (date(2024, 1, 15), date(2024, 1, 17)),
+        (date(2024, 1, 15), date(2024, 2, 15)),
+        (date(2024, 1, 15), date(2025, 1, 15)),
+    ],
+)
+def test_bootcamp_duration_format(start, end):
+    """Bootcamp Date format tests"""
+    expected_date = (
+        "January 15 - 17, 2024"
+        if start.year == end.year and start.month == end.month
+        else "January 15 - February 15, 2024"
+        if start.year == end.year
+        else "January 15, 2024 - January 15, 2025"
+    )
+    assert bootcamp_duration_format(start, end) == expected_date
```

Comment:
You can move this expected date in the parameters of the test. 
e.g. in this case you could make it simpler by removing it from here and your test would roughly look like:

```suggestion
@pytest.mark.parametrize(
    "start, end, expected_format",
    [
        [date(2024, 1, 15), date(2024, 1, 17), "January 15 - 17, 2024"],
        [date(2024, 1, 15), date(2024, 2, 15), "January 15 - February 15, 2024"],
        [date(2024, 1, 15), date(2025, 1, 15), "January 15, 2024 - January 15, 2025"],
        [date(2024, 1, 15), date(2025, 2, 15), "January 15, 2024 - February 15, 2025"],
    ],
)
def test_bootcamp_duration_format(start, end, expected_format):

    """bootcamp_duration_format should format the duration correctly """
    assert bootcamp_duration_format(start, end) == expected_format
```